test/fec_utest.cpp

```diff
@@ -27,156 +27,119 @@
-    }
-
-    void test_fec_rs_gf2n_fft()
-    {
-        std::cout << "Test fec::RsGf2nFft with sizeof(T)=" << sizeof(T) << '\n';
-        for (size_t wordsize = 1; wordsize <= sizeof(T); wordsize *= 2) {
-            test_fec_rs_gf2n_fft_with_wordsize(wordsize);
-        }
-    }
-    void test_fec_rs_gf2n_fft_with_wordsize(int wordsize)
-    {
-        std::cout << "Test fec::RsGf2nFft with wordsize=" << wordsize << '\n';
-
-        unsigned n_data = 3;
-        unsigned n_parities = 3;
-
-        quadiron::fec::RsGf2nFft<T> fec(wordsize, n_data, n_parities);
-        run_test(&fec, fec.n, n_data, n_data + n_parities);
-    }
-
-    void test_fec_rs_gf2n_fft_add()
-    {
-        std::cout << "Test fec::RsGf2nFftAdd with sizeof(T)=" << sizeof(T)
-                  << '\n';
-        for (size_t wordsize = 1; wordsize <= sizeof(T); wordsize *= 2)
-            test_fec_rs_gf2n_fft_add_with_wordsize(wordsize);
-    }
-    void test_fec_rs_gf2n_fft_add_with_wordsize(int wordsize)
-    {
-        std::cout << "Test fec::RsGf2nFftAdd with wordsize=" << wordsize
-                  << '\n';
-
-        unsigned n_data = 3;
-        unsigned n_parities = 3;
-
-        quadiron::fec::RsGf2nFftAdd<T> fec(wordsize, n_data, n_parities);
-        run_test(&fec, fec.n, n_data, n_data + n_parities);
-    }
-
-    void test_fec_rs_gfp_fft()
-    {
-        std::cout << "Test fec::RsGfpFft with sizeof(T)=" << sizeof(T) << '\n';
-        for (size_t word_size = 1; word_size <= 4 && word_size < sizeof(T);
-             word_size *= 2) {
-            test_fec_rs_gfp_fft_with_word_size(word_size);
-        }
-    }
-    void test_fec_rs_gfp_fft_with_word_size(int word_size)
-    {
-        std::cout << "Test fec::RsGfpFft with word size=" << word_size << '\n';
-
-        unsigned n_data = 3;
-        unsigned n_parities = 3;
-
-        quadiron::fec::RsGfpFft<T> fec(word_size, n_data, n_parities);
-        run_test(&fec, fec.n, n_data, n_data + n_parities, true);
-    }
+    const unsigned n_data = 3;
+    const unsigned n_parities = 3;
 
     void run_test(
-        quadiron::fec::FecCode<T>* fec,
-        int n,
+        fec::FecCode<T>& fec,
         int n_data,
         int code_len,
         bool props_flag = false)
     {
-        const quadiron::gf::Field<T>* gf = &(fec->get_gf());
+        const quadiron::gf::Field<T>* gf = &(fec.get_gf());
 
         quadiron::vec::Vector<T> v(*gf, n_data);
-        quadiron::vec::Vector<T> _v(*gf, n);
+        quadiron::vec::Vector<T> _v(*gf, fec.n);
         quadiron::vec::Vector<T> _v2(*gf, n_data);
         quadiron::vec::Vector<T> f(*gf, n_data);
         quadiron::vec::Vector<T> v2(*gf, n_data);
         quadiron::vec::Vector<T> v_p(*gf, n_data);
         std::vector<int> ids;
-        for (int i = 0; i < code_len; i++)
+
+        for (int i = 0; i < code_len; i++) {
             ids.push_back(i);
+        }
+
         std::vector<quadiron::Properties> props(code_len);
         for (int j = 0; j < 1000; j++) {
```

Comment:
@lamphamsy why are we iterating 1000 times?

It's the case in several tests and it makes some tests slow, is there a reason why we are doing this?

Reply:
Generally, we should test for different input data. Here, we test for different input data `v` and received fragments lists

Reply:
Ok, I see.

That being said, why 1000? Is there some statistical reason behind it? Or would 100 be just as fine?

Reply:
Personally I think 100 is a bit small. But if the running time for 1000 is too long, we could choose a smaller one.

Reply:
Ok, we can keep 1000 for now and see if it's a problem later (as we add more tests)